GVFS/GVFS.Virtualization/Projection/GitIndexProjection.GitIndexEntry.cs

```diff
@@ -1,6 +1,9 @@
 ﻿using GVFS.Common;
+using System;
+using System.IO;
 using System.Linq;
-
+using System.Text;
+
 namespace GVFS.Virtualization.Projection
 {
     public partial class GitIndexProjection

```

Comment:
The changes in this file feel very complicated. Can you describe the problem that was being solved here?

Reply:
Yea this code is intentionally not thread-safe, because the index parsing always happens on one thread and we didn't want to take that perf hit. Let me try to better understand why there's a race from the background thread and see if I can suggest anything to simplify this down.

Reply:
As discussed offline, I will try removing all the new checks from `ParsePath` so that we don't impact the hot-path.  Stay tuned for that commit and let me know what you think once it's up.